### PR TITLE
Port host_power to apypie

### PR DESCRIPTION
**Pull request by bagasse**
* Port host_power to apypie
* Add some state aliases supported by foreman
* Make hostname an alias of name

### CODE REVIEW

File: plugins/modules/foreman_host_power.py [hunk 1]
```diff
@@ -87,41 +92,34 @@
...
 
-    entity = find_host(module, host_dict['hostname'], failsafe=True)
-
-    if state == 'state':
-        power_state = query_power_state(module, entity)
-        module.exit_json(changed=False, power_state=power_state)
+    entity = module.find_resource_by_name('hosts', name=entity_dict['name'], failsafe=False, thin=True)
 
+    params = {'id': entity_dict['name']}
+    _, power_state = module.resource_action('hosts', 'power_status', params=params)
```
Comment (mdellweg) [inline plugins/modules/foreman_host_power.py]: I guess not...

File: plugins/modules/foreman_host_power.py [hunk 2]
```diff
@@ -87,41 +92,34 @@
...
-    )
-
-    from ansible.module_utils.ansible_nailgun_cement import (
-        find_host,
-        ForemanEntityAnsibleModule,
-        naildown_power_state,
-        query_power_state,
-    )
-except ImportError:
-    pass
```
Comment (mdellweg) [inline plugins/modules/foreman_host_power.py]: parameter_entity_spec is not used

**Comment by bagasse** [comment]
@mdellweg, thanks for the review. Fixed both of your comments.

File: plugins/modules/foreman_host_power.py (hunk 1 above)
Comment (mdellweg) [reply plugins/modules/foreman_host_power.py]: hmm, this mobile client dosn't seem to comment on quite the right line. Glad you understood me anyway.

File: tests/test_playbooks/host_power.yml [hunk 3]
```diff
@@ -1,32 +1,80 @@
...
   - include: tasks/host_power.yml
     vars:
+      host_power_name: "{{ host_name }}"
+      host_power_state: "start"
+      expected_change: false
+
+  # Sometime host power state change is not reflected instantaneously
+  # and make this playbook fail when recording tests
+  - pause:
+      seconds: 5
```
Comment (mdellweg) [inline tests/test_playbooks/host_power.yml]: I think in syc, there is an example for not waiting while not recording.

Comment (bagasse) [reply tests/test_playbooks/host_power.yml]: @mdellweg, thanks for the pointer !

**Pull request merged**